docs/cookbook/adding-new-attribute-to-an-entity.md

```diff
@@ -176,37 +176,49 @@ Shopsys\FrameworkBundle\Model\Product\ProductFactoryInterface: '@Shopsys\ShopBun
 
 9. Create new `ProductDataFactory` in the same namespace as your entity
 by extending [`Shopsys\FrameworkBundle\Model\Product\ProductDataFactory`](../../packages/framework/src/Model/Product/ProductDataFactory.php)
-and overwrite the `createDefault()` method. 
+and overwrite `create()` and `createFromProduct()` methods. Or you can create an independent class by implementing
+[`Shopsys\FrameworkBundle\Model\Product\ProductDataFactoryInterface`](../../packages/framework/src/Model/Product/ProductDataFactoryInterface.php).
 
-*Note: There is no interface for `ProductDataFactory` like it was in the previous step for `ProductFactory`. 
-This issue will be addressed in near future.*
 ```php
 <?php
 
 namespace Shopsys\ShopBundle\Model\Product;
 
+use Shopsys\FrameworkBundle\Model\Product\Product;
+use Shopsys\FrameworkBundle\Model\Product\ProductData as BaseProductData;
 use Shopsys\FrameworkBundle\Model\Product\ProductDataFactory as BaseProductDataFactory;
 
 class ProductDataFactory extends BaseProductDataFactory
 {
     /**
+     * @param \Shopsys\ShopBundle\Model\Product\Product $product
      * @return \Shopsys\ShopBundle\Model\Product\ProductData
      */
-    public function createDefault()
+    public function createFromProduct(Product $product): BaseProductData
```

Comment:
as mentioned before, using `BaseProduct` alias would be more readable in my opinion